erlang/pop_protocol/pop_protocol.erl

```diff
@@ -98,10 +109,14 @@ add_one_block(ProtocolData, Block, CurrentTime)
 
     %% (OPTIONAL) check sequence of different verifiers
 
-    %% Look at transactions
-    %% verify signature's correctness
-    %% compare ChainId
+    %% Verify transactions
+    lists:map(fun(T) -> check_transaction_correctness(T, MainChainId) end, BlockTransactionsList),
+
+    %% add this block to tree_data 
+    %% this can trigger errors if the block if poorly formed
+    %% also fails if block is orphan or already exists
+    TD1 = blocktree:add_new_block(Block, TD0),
 
-    pass.
+    TD1.
```

Comment:
what does TD0/TD1 stand for?

Reply:
TD0 is initial state
TD1 is the modified final state 
(function modifies state by adding new block)

Reply:
what does TD stand for? Tower Defense :o?

Reply:
Tree Data lol
Actually it's incorrect! Should update the whole state not just tree data

Reply:
Such code review O: